tests/browsercontext-proxy.spec.ts

```diff
@@ -115,6 +115,26 @@ it('should use proxy for second page', async ({contextFactory, server}) => {
   await context.close();
 });
 
+it('should use proxy for https urls', async ({ contextFactory, server, httpsServer, proxyServer }) => {
+  httpsServer.setRoute('/target.html', async (req, res) => {
+    res.end('<html><title>Served by the proxy</title></html>');
+  });
+  let connectedToProxy = false;
+  proxyServer.onConnect(req => {
+    req.url = `localhost:${httpsServer.PORT}`;
+    connectedToProxy = true;
+  });
+  const context = await contextFactory({
+    ignoreHTTPSErrors: true,
+    proxy: { server: `localhost:${proxyServer.PORT}` }
```

Comment:
Should we migrate more/all tests here to the new proxy?

Reply:
Probably, will have a look.